milli/src/search/criteria/attribute.rs

```diff
@@ -0,0 +1,715 @@
+            None => (self.level.saturating_sub(1), None),
+        };
+
+        let left_interval = self.accumulator.get(self.interval_to_skip).map(|opt| opt.as_ref().map(|(left, _, _)| *left)).flatten();
+        let mut inner = Vec::with_capacity(self.inner.len());
+        for word_level_iterator in self.inner.iter() {
+            inner.push(word_level_iterator.dig(ctx, &level, left_interval)?);
+        }
+
+        Ok(Self {parent, inner, level, accumulator: vec![], parent_accumulator: vec![], interval_to_skip: 0})
+    }
+
+    fn inner_next(&mut self, level: TreeLevel) -> heed::Result<Option<(u32, u32, RoaringBitmap)>> {
+        let mut accumulated: Option<(u32, u32, RoaringBitmap)> = None;
+        let u8_level = Into::<u8>::into(level);
+        let interval_size = 4u32.pow(u8_level as u32);
+        for wli in self.inner.iter_mut() {
+            let wli_u8_level = Into::<u8>::into(wli.level.clone());
+            let accumulated_count = 4u32.pow((u8_level - wli_u8_level) as u32);
+            for _ in 0..accumulated_count {
+                if let Some((next_left, _, next_docids)) =  wli.next()? {
+                    accumulated = match accumulated.take(){
+                        Some((acc_left, acc_right, mut acc_docids)) => {
+                            acc_docids |= next_docids;
+                            Some((acc_left, acc_right, acc_docids))
+                        },
+                        None => Some((next_left, next_left + interval_size, next_docids)),
+                    };
+                }
+            }
+        }
+
+        Ok(accumulated)
+    }
+
+    /// return the next meta-interval created from inner WordLevelIterators,
+    /// and from eventual chainned QueryLevelIterator.
+    fn next(&mut self, allowed_candidates: &RoaringBitmap, tree_level: TreeLevel) -> heed::Result<Option<(u32, u32, RoaringBitmap)>> {
+        let parent_result = match self.parent.as_mut() {
+            Some(parent) => {
+                Some(parent.next(allowed_candidates, tree_level)?)
+            },
+            None => None,
+        };
+
+        match parent_result {
+            Some(parent_next) => {
+                let inner_next = self.inner_next(tree_level)?;
+                self.interval_to_skip += self.accumulator.iter().zip(self.parent_accumulator.iter()).skip(self.interval_to_skip).take_while(|current| {
+                    match current {
+                        (Some((_, _, inner)), Some((_, _, parent))) => {
+                            inner.is_disjoint(allowed_candidates) && parent.is_empty()
+                        },
+                        (Some((_, _, inner)), None) => {
+                            inner.is_disjoint(allowed_candidates)
+                        },
+                        (None, Some((_, _, parent))) => {
+                            parent.is_empty()
+                        },
+                        (None, None) => true,
+                    }
+                }).count();
```

Comment:
Could you please reduce this code block by maybe using multiple variables and computing intermediate values, first?